ghost/tinybird-dedicated/scripts/exec_test.sh

```
@@ -15,44 +57,63 @@ run_test() {
+        rm "$tmpfile"
         return 0
     elif [ $retries -eq $TOTAL_RETRIES ]; then
-        echo "🚨 ERROR: Test $t failed, diff:";
-        diff -B ${t}.result $tmpfile
-        rm $tmpfile
+        echo "🚨 ERROR: Test $t failed, showing differences:"
+        diff -B -u --color -U3 "${t}.result" "$tmpfile"  # Use unified diff format with 3 lines of context
+        rm "$tmpfile"
         return 1
     else
         echo "🚨 ERROR: Test $t failed with bash command exit code $?"
```

Shellcheck (0.10.0)

[warning] This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten.

(SC2319)